lib/Gedmo/Tree/Strategy/ORM/Nested.php

```diff
@@ -534,6 +534,9 @@ public function shiftRL(EntityManager $em, $class, $first, $delta, $root = null)
                 if ($node instanceof Proxy && !$node->__isInitialized__) {
                     continue;
                 }
+                if (get_class($node) !== $class) {
```

Comment:
This won't work as the problem is the `$meta` is always the correct NestedSet entity but the `$node` can be a non-nested-set entity.